tdm-fe/src/app/components/progress/StatusTimeline.tsx

```diff
@@ -0,0 +1,84 @@
+import '~/app/components/progress/scss/StatusTimeline.scss';
+import ErrorOutlineIcon from '@mui/icons-material/ErrorOutline';
+import TimelineConnector from '@mui/lab/TimelineConnector';
+import TimelineContent from '@mui/lab/TimelineContent';
+import TimelineDot from '@mui/lab/TimelineDot';
+import TimelineItem from '@mui/lab/TimelineItem';
+import TimelineSeparator from '@mui/lab/TimelineSeparator';
+import CircularProgress from '@mui/material/CircularProgress';
+import { useMemo } from 'react';
+import type { TimelineContentProps } from '@mui/lab/TimelineContent';
+import type { TimelineDotProps } from '@mui/lab/TimelineDot';
+
+export type StatusTimeline = {
+    isRunning?: boolean;
+    isComplet?: boolean;
+    isOnError?: boolean;
+    text: string;
+};
+
+const CustomTimelineDot = ({
+    color,
+    isRunning,
+    isOnError,
+}: {
+    color: TimelineDotProps['color'];
+    isRunning: boolean;
+    isOnError: boolean;
+}) => {
+    if (isRunning) {
+        return (
+            <TimelineDot color={color}>
+                <CircularProgress size={16} />
+            </TimelineDot>
+        );
+    }
+
+    if (isOnError) {
+        return (
+            <TimelineDot color={color} className="status-timeline-error">
+                <ErrorOutlineIcon className="status-timeline-error-icon" />
+            </TimelineDot>
+        );
+    }
+
+    return (
+        <TimelineDot color={color}>
+            <div className="status-timeline-waiting"></div>
+        </TimelineDot>
+    );
+};
+
+const StatusTimeline = ({ isRunning = false, isComplet = false, isOnError = false, text }: StatusTimeline) => {
```

Comment:
<!-- TODO's are small, trivial, but necessary changes. Distinguishing todo comments from issues: or suggestions: helps direct the reader's attention to comments requiring more involvement.  -->
**todo:** rename `isComplet` to `isComplete`